pageserver/src/tenant.rs

```diff
@@ -2812,7 +2812,50 @@ impl Tenant {
         cancel: &CancellationToken,
         ctx: &RequestContext,
     ) -> anyhow::Result<Vec<Arc<Timeline>>> {
-        // grab mutex to prevent new timelines from being created here.
+        let timelines = self
+            .timelines
+            .lock()
+            .unwrap()
+            .values()
+            .filter(|tl| match target_timeline_id.as_ref() {
+                Some(required) => &tl.timeline_id == required,
```

Comment:
No idea why you call it required`, it's confusing to me.

Reply:
What would you call it? It is used to filter the timelines. `only`?

Reply:
Well, `target` is quite obvious.